server/src/modules/report-comments/report-comments.service.ts

```diff
@@ -6,19 +6,23 @@ import { ObjectIdScalar } from '../common/graphql-scalars/object-id.scalar';
 import { AddReportCommentInput } from './models/add-report-comment.input';
 import { EditReportCommentInput } from './models/edit-report-comment.input';
 import { ReportComment } from './models/report-comment.schema';
-
+import { ErrorResponse } from '../common/graphql-generic-responses/error-response.model';
+import { ObjectId } from 'bson';
+import { FindAllCommentsInput } from './models/find-all-comments.input';
 @Injectable()
 export class ReportCommentsService {
   constructor(
     @InjectModel(ReportComment.name)
     private readonly commentModel: ReturnModelType<typeof ReportComment>,
   ) {}
 
-  async findAll(reportId: ObjectIdScalar): Promise<ReportComment[]> {
-    return this.commentModel.find({ reportId }).exec();
+  async findAll(request: FindAllCommentsInput): Promise<ReportComment[]> {
+    return this.commentModel
+      .find({ reportId: request.reportId, answeredTo: request.answeredTo })
+      .exec();
   }
 
-  async findOne(id: ObjectIdScalar): Promise<ReportComment> {
+  async findOne(id: ObjectIdScalar | ObjectId): Promise<ReportComment> {
```

Comment:
Why?

Reply:
move it to common `type ObjectIdAdapt` or sth